web/packages/teleport/src/Console/DocumentSsh/DocumentSsh.tsx

```diff
@@ -54,14 +54,15 @@ function DocumentSsh({ doc, visible }: PropTypes) {
   const { tty, status, closeDocument, session } = useSshSession(doc);
   const [showSearch, setShowSearch] = useState(false);
 
-  const ttyMfa = useMfaEmitter(tty);
-  const ftMfa = useMfa({
-    isMfaRequired: ttyMfa.required,
+  const mfa = useMfaEmitter(tty, {
+    // The MFA requirement will be determined by whether we do/don't get
+    // an mfa challenge over the event emitter at session start.
+    isMfaRequired: false,
```

Comment:
if this field `isMfaRequired` is optional, it will default to false anyway,

besides a test, i dont see this field ever being used. is it needed? (i dont see it set true anywhere outside of a test)

Reply:
It defaults to undefined if it isn't passed in

Reply:
Sorry, I meant defaults to a falsey value, which we can just parse to a Boolean

Reply:
We compare to false directly in order to get the behavior of skipping MFA / MFA check when we are certain mfa is not necessary:

https://github.com/gravitational/teleport/blob/ce976bbe9d83c97589e8db32e825c83c0e64a6ba/web/packages/teleport/src/lib/useMfa.ts#L76

Reply:
Thats the state value tho, im talking about the `isMfaRequired` prop. because its optional, you can give it a default value
```js
export function useMfa({ req, isMfaRequired = false }: MfaProps): MfaState {
```
you dont have to pass `false` for the prop

Reply:
We're initializing the state from the prop now so that we don't need any `useEffect` trickery - https://github.com/gravitational/teleport/blob/ce976bbe9d83c97589e8db32e825c83c0e64a6ba/web/packages/teleport/src/lib/useMfa.ts#L58

Reply:
Sure whatever the default should be, I was just suggesting we could have that default set in the destructuring.